src/Gamecube.h

```diff
@@ -88,6 +88,180 @@ typedef union{
+        uint8_t ddown : 1;
+        uint8_t dup : 1;
+        uint8_t z : 1;
+        uint8_t r : 1;
+        uint8_t l : 1;
+        uint8_t high1 : 1;
+
+        // 3rd-8th data byte
+        uint8_t xAxis;
+        uint8_t yAxis;
+        uint8_t cxAxis;
+        uint8_t cyAxis;
+        /*
+        uint8_t left;
+        uint8_t right;
+        */
+        uint8_t left : 4;
+        uint8_t right : 4;
+        uint8_t analogA : 4;
+        uint8_t analogB : 4;
+    } Mode0;
+
+    struct {
+        // first data byte (bitfields are sorted in LSB order)
+        uint8_t a : 1;
+        uint8_t b : 1;
+        uint8_t x : 1;
+        uint8_t y : 1;
+        uint8_t start : 1;
+        uint8_t origin : 1; // Indicates if GetOrigin(0x41) was called (LOW)
+        uint8_t errlatch : 1;
+        uint8_t errstat : 1;
+
+        // second data byte
+        uint8_t dleft : 1;
+        uint8_t dright : 1;
+        uint8_t ddown : 1;
+        uint8_t dup : 1;
+        uint8_t z : 1;
+        uint8_t r : 1;
+        uint8_t l : 1;
+        uint8_t high1 : 1;
+
+        // 3rd-8th data byte
+        uint8_t xAxis;
+        uint8_t yAxis;
+        uint8_t cxAxis : 4;
+        uint8_t cyAxis : 4;
+        uint8_t left;
+        uint8_t right;
+        uint8_t analogA : 4;
+        uint8_t analogB : 4;
+    } Mode1;
+
+    struct {
+        // first data byte (bitfields are sorted in LSB order)
+        uint8_t a : 1;
+        uint8_t b : 1;
+        uint8_t x : 1;
+        uint8_t y : 1;
+        uint8_t start : 1;
+        uint8_t origin : 1; // Indicates if GetOrigin(0x41) was called (LOW)
+        uint8_t errlatch : 1;
+        uint8_t errstat : 1;
+
+        // second data byte
+        uint8_t dleft : 1;
+        uint8_t dright : 1;
+        uint8_t ddown : 1;
+        uint8_t dup : 1;
+        uint8_t z : 1;
+        uint8_t r : 1;
+        uint8_t l : 1;
+        uint8_t high1 : 1;
+
+        // 3rd-8th data byte
+        uint8_t xAxis;
+        uint8_t yAxis;
+        uint8_t cxAxis : 4;
+        uint8_t cyAxis : 4;
+        uint8_t left : 4;
+        uint8_t right : 4;
+        uint8_t analogA;
+        uint8_t analogB;
+    } Mode2;
+
+    struct {
+        // first data byte (bitfields are sorted in LSB order)
+        uint8_t a : 1;
+        uint8_t b : 1;
+        uint8_t x : 1;
+        uint8_t y : 1;
+        uint8_t start : 1;
+        uint8_t origin : 1; // Indicates if GetOrigin(0x41) was called (LOW)
+        uint8_t errlatch : 1;
+        uint8_t errstat : 1;
+
+        // second data byte
+        uint8_t dleft : 1;
+        uint8_t dright : 1;
+        uint8_t ddown : 1;
+        uint8_t dup : 1;
+        uint8_t z : 1;
+        uint8_t r : 1;
+        uint8_t l : 1;
+        uint8_t high1 : 1;
+
+        // 3rd-8th data byte
+        uint8_t xAxis;
+        uint8_t yAxis;
+        uint8_t cxAxis;
+        uint8_t cyAxis;
+        uint8_t analogA;
+        uint8_t analogB;
+    } Mode4;
+} Gamecube_Different_Report_t;
```

Comment:
I think this looks correct. Was there a specific reason why to give it a different name? I think it could make sense to just merge it with the already existing report structure.

Reply:
What about this one? What do you think?

Reply:
By structure I mean `Gamecube_Different_Report_t`. What about merging `Gamecube_Different_Report_t` into `Gamecube_Report_t`? I think a single struct will be sufficient.